### PR TITLE
Drop support for TLSv1.0 and TLSv1.1

### PR DESCRIPTION
**Pull request by JLLeitschuh**
These versions of TLS have known security vulnerabilities.
If a user needs to re-enable these SSL versions, they can do so with the `https.protocols` system property.

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
@bot-gradle test SC

**Comment by JLLeitschuh** [comment]
@bot-gradle test ReadyForMerge

**Comment by JLLeitschuh** [comment]
@bot-gradle test ReadyForMerge

**Comment by JLLeitschuh** [comment]
@bot-gradle help

**Comment by JLLeitschuh** [comment]
@bot-gradle test this

**Comment by JLLeitschuh** [comment]
@bot-gradle test this

**Comment by JLLeitschuh** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by JLLeitschuh** [comment]
@bot-gradle test this

**Comment by JLLeitschuh** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/DeprecatedTLSVersionDependencyResolutionIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            )
+        )
+    }
+
+    def "able to resolve dependencies when the user manually specifies the supported TLS versions using `https.protocols`"() {
+        given:
+        keyStore.enableSslWithServerCert(mavenHttpRepo.server) {
+            it.addExcludeProtocols("TLSv1.2", "TLSv1.3")
+            it.setIncludeProtocols("TLSv1", "TLSv1.1")
+            it.setExcludeCipherSuites()
```
Comment (JLLeitschuh) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/DeprecatedTLSVersionDependencyResolutionIntegrationTest.groovy:70]: Thanks @big-guy for figuring out this last one.

**Pull request merged**